lib/iris/fileformats/grib/message.py

```diff
@@ -150,14 +154,18 @@ def data(self):
                 shape = (grid_section['numberOfDataPoints'],)
             else:
                 shape = (grid_section['Nj'], grid_section['Ni'])
-            proxy = _DataProxy(shape, np.dtype('f8'), np.nan,
+            proxy = _DataProxy(shape, np.dtype('f8'), self.bmdi,
```

Comment:
I needed to add a bmdi as `iris.fileformats.rules._make_cube` requires it (see https://github.com/lbdreyer/iris/blob/c8b497e7a7e94ea526df3a5352f2c2b42cebf1ae/lib/iris/fileformats/rules.py#L909)

I chose nans as that is what we had been using (and I made self.bmdi a property so we could reuse it)